src/ast/passes/semantic_analyser.cpp

```diff
@@ -125,6 +125,18 @@ static bool IsValidVarDeclType(const SizedType &ty)
   return false; // unreachable
 }
 
+// These are special map aggregation types that cannot be assigned
+// to scratch variables or from one map to another e.g. these are both invalid:
+// `@a = hist(10); let $b = @a;`
+// `@a = hist(10); @b = @a;`
+static bool IsValidAssignment(const SizedType &ty, bool is_map)
+{
+  if (is_map && (ty.IsHistTy() || ty.IsLhistTy() || ty.IsStatsTy())) {
```

Comment:
Should this be inverted? So that new types are not automatically assignable by default?

Or maybe adding an `IsAssignable` method to on the `type` itself?

Reply:
> Or maybe adding an IsAssignable method to on the type itself?

I like this and will update. But also think that by default types should be assignable. I don't see us really adding more map aggregation types in the future (seems like something our future standard library should provide).

Reply:
Oh actually, I realized that we need the `is_map` part which is part of the expression so we can't do this on the type itself.

Reply:
Very interesting that avg(), count(), sum(), etc. all work. Intentional or just lucky?

Reply:
Intentional. Min and max also work because we generate single values for all of these by looping over the per-cpu map e.g. https://github.com/bpftrace/bpftrace/pull/3268